File: programming/robot/src/Robot.cpp
#include "Robot.h"
#include <iostream>

string Robot::robotManufactuer = "SIT Robotics";
int Robot::numberOfRobotsExisting = 0;

Robot::Robot(string serial, int numWheels, string LiDARType, bool withCam, int initial_X, int initial_Y) : base(numWheels, LiDARType, withCam, initial_X, initial_Y)
{
    serialNumber = serial;
    numberOfRobotsExisting++;
}

void Robot::ShowManufactuer()
{
    cout << "Manufacturer is: " << robotManufactuer << endl;
}

void Robot::ShowNumberOfRobotsExisting()
{
    cout << "Number of Robots Existing is: " << numberOfRobotsExisting << endl;
}

void Robot::ShowSerialNo()
{
    cout << "Serial Number is: " << serialNumber << endl;
}

File: programming/robot/src/main/java/robotics/Robot.java
package robotics;

public class Robot {
    private static String robotManufacturer = "Robotics Inc.";
    private static int numberofRobotsExisting = 0;
    private String serialNumber;
    public MobileBase base;
    public Robot(String serial, int numwheels, String LiDARType, boolean withCam, int initial_x, int initial_y) {
        serialNumber = serial;
        base = new MobileBase(numwheels, LiDARType, withCam, initial_x, initial_y);
        numberofRobotsExisting++;
    }

    public void ShowManufacturer() {
        System.out.println("Manufacturer is: " + robotManufacturer);        
    }

    public void ShowSerialNo() {
        System.out.println("Serial number is: " + serialNumber);        
    }

    public void ShowNumberofRobotsExisting() {
        System.out.println("Number of robots existing is: " + numberofRobotsExisting);        
    }
}
